### PR TITLE
Add filter_pattern for cloudwatch log groups

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* Add filter_pattern for cloudwatch log groups

## why
* If one log group per account, then it will need to be filtered per app

## references
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_log_subscription_filter#filter_pattern

### CODE REVIEW

**Review by jamengual — APPROVED** [review]

File: variables.tf [hunk 1]
```diff
@@ -133,9 +133,9 @@ variable "s3_bucket_kms_arns" {
 }
 
 variable "cloudwatch_forwarder_log_groups" {
-  type        = map(string)
+  type        = map(map(string))
   description = <<EOT
-    "Map of Cloudwatch Log Groups that the Lambda forwarder will send logs from. For example: { mysql1 = "/aws/rds/maincluster"}"
+    Map of Cloudwatch Log Groups that the Lambda forwarder will send logs from. For example: { mysql1 = { name = "/aws/rds/maincluster", filter_pattern = "" }
```
Comment (aknysh) [inline variables.tf:138]:
```suggestion
    Map of CloudWatch Log Groups with a filter pattern that the Lambda forwarder will send logs from. For example: { mysql1 = { name = "/aws/rds/maincluster", filter_pattern = "" }
```

File: variables.tf [hunk 2]
```diff
@@ -203,3 +203,15 @@ variable "forwarder_vpc_logs_layers" {
   description = "List of Lambda Layer Version ARNs (maximum of 5) to attach to Datadog VPC flow log forwarder lambda function"
   default     = []
 }
+
+variable "forwarder_rds_filter_pattern" {
+  type        = string
+  description = "Filter pattern for lambda forwarder RDS"
```
Comment (aknysh) [inline variables.tf:209]:
```suggestion
  description = "Filter pattern for Lambda forwarder RDS"
```

File: variables.tf [hunk 3]
```diff
@@ -203,3 +203,15 @@ variable "forwarder_vpc_logs_layers" {
...
+  default     = ""
+}
+
+variable "forwarder_vpclogs_filter_pattern" {
+  type        = string
+  description = "Filter pattern for lambda forwarder VPC Logs"
```
Comment (aknysh) [inline variables.tf:215]:
```suggestion
  description = "Filter pattern for Lambda forwarder VPC Logs"
```

**Review by aknysh — COMMENTED** [review]
a few nitpicks

**Review by aknysh — APPROVED** [review]

**Pull request merged**